File: result/metagpt/claude/humaneval/141.c
#include <stdio.h>
#include <string.h>

const char* file_name_check(const char* file_name) {
    int numdigit = 0, numdot = 0;
    int length = strlen(file_name);
    
    if (length < 5) return "No";
    
    char w = file_name[0];
    if (!((w >= 'A' && w <= 'Z') || (w >= 'a' && w <= 'z'))) return "No";
    
    const char* last = file_name + length - 4;
    if (strcmp(last, ".txt") != 0 && strcmp(last, ".exe") != 0 && strcmp(last, ".dll") != 0) return "No";
    
    for (int i = 0; i < length; i++) {
        if (file_name[i] >= '0' && file_name[i] <= '9') numdigit++;
        if (file_name[i] == '.') numdot++;
    }
    
    if (numdigit > 3 || numdot != 1) return "No";
    
    return "Yes";
}

File: result/metagpt/gpt4o/humaneval/141.rs
fn file_name_check(file_name: &str) -> &'static str {
    let length = file_name.len();
    if length < 5 {
        return "No";
    }

    let first_char = file_name.chars().next().unwrap();
    if !first_char.is_alphabetic() {
        return "No";
    }

    let last_four = &file_name[length - 4..];
    if last_four != ".txt" && last_four != ".exe" && last_four != ".dll" {
        return "No";
    }

    let numdigit = file_name.chars().filter(|&ch| ch.is_digit(10)).count();
    let numdot = file_name.chars().filter(|&ch| ch == '.').count();

    if numdigit > 3 || numdot != 1 {
        return "No";
    }

    "Yes"
}

fn main() {
    let file_name = "example.txt";
    let result = file_name_check(file_name);
    println!("File name check result: {}", result);
}